### PR TITLE
fix: error custom edge case

### LINKED ISSUE #322: The custom error addition should handle edge cases
Please add a test about a custom error handler that could be used to invalidate an error by returning nil

adding a custom error handler allow to mess up things, because people may do that in their custom handler if you dont want to support it, the mechanism of error handling should be rewritten to forbid this behavior

_Originally posted by @ccoVeille in https://github.com/go-fuego/fuego/pull/320#discussion_r1899168562_

### PR DESCRIPTION
**Pull request by martinvibes**
closes #322

### CODE REVIEW

**Comment by martinvibes** [comment]
gm @ccoVeille kindly review :)

**Comment by EwenQuim** [comment]
It seems that is solves #366, not #322. So it's a duplicate of #371.

Honestly, as @NueloSE wasn't answering, it does not bother me.

File: option.go [hunk 1]
```diff
@@ -438,3 +439,29 @@ func OptionSecurity(securityRequirements ...openapi3.SecurityRequirement) func(*
...
+
+		// Add middleware to strip trailing slash from requests
+		stripSlashMiddleware := func(next http.Handler) http.Handler {
+			return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if len(r.URL.Path) > 1 {
+					r.URL.Path = strings.TrimRight(r.URL.Path, "/")
+				}
+				next.ServeHTTP(w, r)
+			})
+		}
```
Comment (EwenQuim) [inline option.go:461]: Please do not re-define this middleware each time the option is called. Place the middleware elsewhere :) Also, I think the middleware should be placed in `Server.globalMiddlewares`, see the difference [in the docs](https://go-fuego.github.io/fuego/docs/guides/middlewares).

**Review by EwenQuim — CHANGES_REQUESTED** [review]
Good PR, well structured!

Please **add tests** to show that it works for several patterns, and inform our users of this behaviour in the docs :)

File: server.go [hunk 2]
```diff
@@ -449,3 +449,18 @@ func WithLoggingMiddleware(loggingConfig LoggingConfig) func(*Server) {
...
+//
+//	app := fuego.NewServer(
+//		fuego.WithStripTrailingSlash(),
+//	)
+func WithStripTrailingSlash() func(*Server) {
+	return func(s *Server) {
+		// Add OptionStripTrailingSlash to the default route options
+		s.routeOptions = append(s.routeOptions, OptionStripTrailingSlash())
+	}
+}
```
Comment (EwenQuim) [inline server.go:466]: Very good! With one Server Option, they will be able to have this feature :)

I think it should look like this:

```go
s.routeOptions = append(s.routeOptions, OptionStripTrailingSlash())
s.globalMiddlewares = append(s.globalMiddlewares, stripTrailingSlashesMiddleware)
```

**Comment by martinvibes** [comment]
kindly review @EwenQuim

**Comment by dylanhitt** [comment]
Edited title so I don't keep getting confused 😄

**Comment by martinvibes** [comment]
😅

**Comment by EwenQuim** [comment]
> Please **add tests** to show that it works for several patterns, and inform our users of this behaviour in the docs :)
> — @EwenQuim

> kindly review 
> — @martinvibes after adding absolutely 0 tests

mmmh 🤔

Also, what about [this comment](https://github.com/go-fuego/fuego/pull/379#discussion_r1937594117) and [this one](https://github.com/go-fuego/fuego/pull/379#discussion_r1937562259) ?

Seems that it wasn't really ready lol ;)

**Comment by EwenQuim** [comment]
@martinvibes any news on this PR? Do you need some help?

**Comment by dylanhitt** [comment]
@martinvibes any word on this?

@EwenQuim if we don't hear anything by thursday I'll go ahead a finish this.

File: server.go (hunk 2 above)
Comment (dylanhitt) [reply server.go]: Yeah I'm a bit confused here as well. Why on the globalMiddlewares? That seems more like a thing for cors? At least with just 

```
s.routeOptions = append(s.routeOptions, OptionStripTrailingSlash())
```

You can apply this at both the server and group level

Comment (dylanhitt) [reply server.go]: Ope nevermind. I see now this literally isn't possible unless you register stripping the path at the global level. Got it. I'll make a PR on how I think this should look.

**Comment by EwenQuim** [comment]
Done in #432 ! 

Thank you for the work @martinvibes, it was really interesting to launch the other PR :)
